### PR TITLE
⚡️ Small optimization to GraphQL::Schema::InputObject#to_h

### PR DESCRIPTION
**Pull request by danielvdao**
I think this is a small optimization to the to_h call. I noticed it while stumbling through source code for a project I am working on. `#merge!` will avoid creating extra hash objects. I think this would be faster if the argument type had a lot of fields / nested fields on it.

### CODE REVIEW

**Comment by danielvdao** [comment]
@rmosolgo I don't this is related to an issue, it's just something I noticed. If it's not worth it - feel free to ignore 🙂 . Also not sure how to trigger tests, maybe you can do it?

File: lib/graphql/schema/input_object.rb [hunk 1]
```diff
@@ -54,24 +62,12 @@ def initialize(arguments = nil, ruby_kwargs: nil, context:, defaults_used:)
...
-
-      # Ruby-like hash behaviors, read-only
-      def_delegators :@ruby_style_hash, :keys, :values, :each, :map, :any?, :empty?
-
       def to_h
-        @ruby_style_hash.inject({}) do |h, (key, value)|
-          h.merge(key => unwrap_value(value))
+        @ruby_style_hash.reduce({}) do |h, (key, value)|
+          h.merge!(key => unwrap_value(value))
         end
```
Comment (rmosolgo) [inline lib/graphql/schema/input_object.rb:68]: Interesting... it looks like the implementation here is the same as `unwrap_value` below. (It was the same before, too.) What do you think of DRYing it? 

```suggestion
        unwrap_value(@ruby_style_hash)
```

File: lib/graphql/schema/input_object.rb [hunk 2]
```diff
@@ -54,24 +62,12 @@ def initialize(arguments = nil, ruby_kwargs: nil, context:, defaults_used:)
...
       end
-
-      def to_hash
-        to_h
-      end
+      alias to_hash to_h
```
[the remark this answers is not in the document]
Comment (danielvdao) [reply lib/graphql/schema/input_object.rb]: Oh that's kind of funny - I was actually trying to track down the difference between `alias_method` vs. `alias` and found a lot of different articles relating to this. I did the change suggested 👍🏽 .

File: lib/graphql/schema/input_object.rb (hunk 1 above)
Comment (danielvdao) [reply lib/graphql/schema/input_object.rb]: Makes sense 🙂

**Comment by danielvdao** [comment]
@rmosolgo This is a little bigger now and I'm happy to drop back to https://github.com/rmosolgo/graphql-ruby/pull/3524/commits/e79945be5414709a510b482803a2333057d308e3. I figured I would try to find all `alias` and -> to `alias_method`. Thoughts?

**Comment by danielvdao** [comment]
I also had a lot of trouble running the test suite locally even after doing `bundle install`, so I haven't had any success on that front, which reduces the feedback loop locally 😖 .

**Comment by danielvdao** [comment]
@rmosolgo quick bump on this as I'm sure you have other things going on! Just wanted to check in and see if there's anything I can do to get this small PR through for us 🙇🏽‍♂️

**Comment by rmosolgo** [comment]
Thanks for addressing this! If you care to share any of the details about what didn't work, I'd be happy to help debug. Thanks for the bump :+1:

**Comment by danielvdao** [comment]
@rmosolgo sounds good. I'd love to try and contribute more as time goes on, so I think this will be worth it to reduce feedback loop on PRs 🙂. I'm happy to move this on Slack too since this may get lost in the weeds, so let me know.

I tried to initially run `$ bundle install` and then `bundle exec rspec spec` for each of these steps and got an error:

```
bundler: failed to load command: rspec (/Users/daniel.dao/.rbenv/versions/2.5.5/bin/rspec)
Traceback (most recent call last):
	17: from /Users/daniel.dao/.rbenv/versions/2.5.5/bin/bundle:23:in `<main>'
	16: from /Users/daniel.dao/.rbenv/versions/2.5.5/bin/bundle:23:in `load'
	15: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/exe/bundle:37:in `<top (required)>'
	14: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/friendly_errors.rb:130:in `with_friendly_errors'
	13: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/exe/bundle:49:in `block in <top (required)>'
	12: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/cli.rb:24:in `start'
	11: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	10: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/cli.rb:30:in `dispatch'
	 9: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	 8: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	 7: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	 6: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/cli.rb:494:in `exec'
	 5: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/cli/exec.rb:28:in `run'
	 4: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/cli/exec.rb:63:in `kernel_load'
	 3: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/cli/exec.rb:63:in `load'
	 2: from /Users/daniel.dao/.rbenv/versions/2.5.5/bin/rspec:23:in `<top (required)>'
	 1: from /Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/rubygems_integration.rb:390:in `block in replace_bin_path'
/Users/daniel.dao/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/bundler-2.2.11/lib/bundler/rubygems_integration.rb:362:in `block in replace_bin_path': can't find executable rspec for gem rspec-core. rspec-core is not currently included in the bundle, perhaps you meant to add it to your Gemfile? (Gem::Exception)
```

Then I tried to add `rspec` to the `graphql.gemspec` file and got this error:

```
Failure/Error: require 'rails/generators'

LoadError:
  cannot load such file -- rails/generators
# ./lib/generators/graphql/type_generator.rb:2:in `require'
# ./lib/generators/graphql/type_generator.rb:2:in `<top (required)>'
# ./lib/generators/graphql/scalar_generator.rb:2:in `require'
# ./lib/generators/graphql/scalar_generator.rb:2:in `<top (required)>'
# ./spec/integration/rails/generators/graphql/scalar_generator_spec.rb:3:in `require'
# ./spec/integration/rails/generators/graphql/scalar_generator_spec.rb:3:in `<top (required)>'

An error occurred while loading ./spec/integration/rails/generators/graphql/union_generator_spec.rb. - Did you mean?
                    rspec ./spec/integration/rails/generators/graphql/mutation_generator_spec.rb
                    rspec ./spec/integration/rails/generators/graphql/enum_generator_spec.rb
                    rspec ./spec/integration/rails/generators/graphql/install_generator_spec.rb

Failure/Error: require 'rails/generators'

LoadError:
  cannot load such file -- rails/generators
# ./lib/generators/graphql/type_generator.rb:2:in `require'
# ./lib/generators/graphql/type_generator.rb:2:in `<top (required)>'
# ./lib/generators/graphql/union_generator.rb:2:in `require'
# ./lib/generators/graphql/union_generator.rb:2:in `<top (required)>'
# ./spec/integration/rails/generators/graphql/union_generator_spec.rb:3:in `require'
# ./spec/integration/rails/generators/graphql/union_generator_spec.rb:3:in `<top (required)>'

An error occurred while loading ./spec/integration/rails/graphql/query/variables_spec.rb.
Failure/Error: if ActionPack::VERSION::MAJOR > 3
```

Then I added `rails` and got:

```
NameError:
  uninitialized constant BaseGeneratorTest
# ./spec/integration/rails/generators/graphql/union_generator_spec.rb:5:in `<top (required)>'

An error occurred while loading ./spec/integration/rails/graphql/query/variables_spec.rb.
Failure/Error: if ActionPack::VERSION::MAJOR > 3

NameError:
  uninitialized constant ActionPack
```

No success otherwise 😢

**Comment by danielvdao** [comment]
Oh actually, I looked in the `Rakefile` after seeing past CI builds. It's `bundle exec rake test` 🤦🏽 . Thanks @rmosolgo and sorry about that.

**Pull request merged**